util/rands/cheap_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package rands
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+
+	exprand "golang.org/x/exp/rand"
+)
+
+var (
+	seed          uint64 = 8729831
+	numDraw              = 100
+	numGoroutines        = 5000
+	job                  = make(chan func(), 2<<20)
+	res                  = make(chan struct{}, 2<<20)
+)
+
+func init() {
+	for i := 0; i < numGoroutines; i++ {
+		go func() {
+			for {
+				f := <-job
+				f()
+				res <- struct{}{}
+			}
+		}()
+	}
+}
```

Comment:
A bit of an explanation of what this is doing would be good, since I don't think I know why we're launching 5000 goroutines

Reply:
Oh yeah, I pulled this in from my gist, we don't want this statically, i made a worker pool and plumbed it in to each benchmark.